openid-4-verifiable-presentations-1_0.md

```diff
@@ -1492,9 +1492,139 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+This section defines a profile of OpenID4VP for use with the W3C Digital Credentials API [@!w3c.digital_credentials_api].
+
+The W3C Digital Credentials API defines a Web Platform API which allows web sites acting as Verifiers
+to request the presentation of Verifiable Credentials. The API itself does not define a Credential exchange protocol
+but can be used with multiple protocols. The Web Platform, working in conjunction with other layers, such as the app platform/operating system, and based on the permission of the End-User, will send the request data along with the web origin of the Verifier to the End-User's chosen Wallet.
+
+This OpenID4VP profile utilizes the mechanisms of the W3C Digital Credentials API while also allowing to leverage advanced security features of OpenID4VP, if needed. It also defines the OpenID4VP request parameters that MAY be used with the W3C Digital Credentials API.
+
+The Digital Credentials API offers several advantages for implementers of both Verifiers and Wallets. 
+
+Firstly, the API serves as a privacy-preserving alternative to invoking Wallets via URLs, particularly custom URL schemes. The underlying app platform will only invoke a Wallet if the user confirms the request based on contextual information about the credential request and the requestor (Verifier). 
+
+Secondly, the session with the user will always continue in the initial context, typically a browser tab, when the request has been fulfilled (or aborted), which results in an improved user experience.
+
+Thirdly, cross-device requests benefit from the use of secure transports with proximity checks, which are handled by the OS platform, e.g., using FIDO CTAP 2.2 with hybrid transports.
+
+And lastly, as part of the request, the Wallet is provided with information about the Verifier's origin as authenticated by the user agent, which is important for phishing resistance.  
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `openid4vp` for this profile.
+
+## Request {#browser_api_request}
+
+The `request` member of the W3C Digital Credentials API [@!w3c.digital_credentials_api] contains an OpenID4VP Authorization Request, where every OpenID4VP Authorization Request parameter is represented as a top-level JavaScript object member.
+
+The following is a non-normative example of how the W3C Digital Credentials API can be used with an unsigned OpenID4VP request when advanced security features of OpenID4VP are not used:
+
+```js
+if ('DigitalCredential' in window) {
+  const credential = await navigator.identity.get({
+    digital: {
+      providers: [{
+        protocol: "openid4vp",
+        request:  {
+          response_type: "vp_token",
+          nonce: "n-0S6_WzA2Mj",
+          client_metadata: {...},
+          presentation_definition: {...}
+        }
+      }]
+    }
+  });
+} else {
+// fallback to other invocation mechanisms
+}
+```
```

Comment:
```suggestion
```js
try {
  const credential = await navigator.identity.get({
    digital: {
      providers: [{
        protocol: "urn:openid.net:oid4vp",
        request:  {
          response_type: "vp_token",
          nonce: "n-0S6_WzA2Mj",
          client_metadata: {...},
          presentation_definition: {...}
        }
      }]
    }
  });
} catch (err) {
  // Handle errors and/or fallback to other invocation mechanisms
}
```
```